src/lib/commandBarConfigs/applicationCommandConfig.ts

```diff
@@ -1,19 +1,100 @@
+import type { RequestedKCLFile } from '@src/machines/systemIO/utils'
 import { SystemIOMachineEvents } from '@src/machines/systemIO/utils'
 import { isDesktop } from '@src/lib/isDesktop'
-import { kclSamplesManifestWithNoMultipleFiles } from '@src/lib/kclSamples'
-import { getUniqueProjectName } from '@src/lib/desktopFS'
 import {
-  FILE_EXT,
-  IS_ML_EXPERIMENTAL,
-  ML_EXPERIMENTAL_MESSAGE,
-} from '@src/lib/constants'
+  everyKclSample,
+  findKclSample,
+  kclSamplesManifestWithNoMultipleFiles,
+} from '@src/lib/kclSamples'
+import { getUniqueProjectName } from '@src/lib/desktopFS'
+import { IS_ML_EXPERIMENTAL, ML_EXPERIMENTAL_MESSAGE } from '@src/lib/constants'
 import toast from 'react-hot-toast'
 import { reportRejection } from '@src/lib/trap'
 import { relevantFileExtensions } from '@src/lang/wasmUtils'
 import { getStringAfterLastSeparator, webSafePathSplit } from '@src/lib/paths'
+import { FILE_EXT } from '@src/lib/constants'
+
+function onSubmitKCLSampleCreation({
+  sample,
+  kclSample,
+  uniqueNameIfNeeded,
+  systemIOActor,
+}: {
+  sample: any
+  kclSample: ReturnType<typeof findKclSample>
+  uniqueNameIfNeeded: any
+  systemIOActor: ActorRefFrom<typeof systemIOMachine>
+}) {
+  if (!kclSample) {
+    toast.error('The command could not be submitted, unable to find Zoo sample')
+    return
+  }
+  const pathParts = webSafePathSplit(sample)
+  const projectPathPart = pathParts[0]
+  const files = kclSample.files
+
+  const filePromises = files.map((file) => {
+    const sampleCodeUrl =
+      (isDesktop() ? '.' : '') +
+      `/kcl-samples/${encodeURIComponent(
+        projectPathPart
+      )}/${encodeURIComponent(file)}`
+    return fetch(sampleCodeUrl).then((response) => {
+      return {
+        response,
+        file,
+        projectName: projectPathPart,
+      }
+    })
+  })
+
+  const requestedFiles: RequestedKCLFile[] = []
+  // If any fetches fail from the KCL Code download we will instantly reject
+  // No cleanup required since the fetch response is in memory
+  // TODO: Try to catch if there is a failure then delete the root folder and show error
+  Promise.all(filePromises)
+    .then(async (responses) => {
+      for (let i = 0; i < responses.length; i++) {
+        const response = responses[i]
+        const code = await response.response.text()
+        requestedFiles.push({
+          requestedCode: code,
+          requestedFileName: response.file,
+          requestedProjectName: uniqueNameIfNeeded,
+        })
+      }
+
+      if (requestedFiles.length === 1) {
+        /**
+         * Navigates to the single file that could be renamed on disk for duplicates
+         */
+        const folderNameBecomesKCLFileName = projectPathPart + FILE_EXT
+        systemIOActor.send({
+          type: SystemIOMachineEvents.importFileFromURL,
+          data: {
+            requestedProjectName: requestedFiles[0].requestedProjectName,
+            requestedFileNameWithExtension: folderNameBecomesKCLFileName,
```

Comment:
nit: this is the bit of logic I was saying if `data.method === "newProject"` we probably don't need to reassign this file name right? so then the user gets something like `/bottle-1/main.kcl` matching the KCL sample as closely as possible

Reply:
I thought we said we wanted `bottle/bottle.kcl`?

Reply:
Yeah this is the thing we went back and forth on 😅